specs/monitoring_integration.adoc

```diff
@@ -0,0 +1,321 @@
+= Spec - Introducing a new component monitoring-integration
+
+Remove existing performance monitoring component and replace it with a new
+monitoring-integration component.
+
+
+== Problem description
+
+The new monitoring-integration component would primarily focus on dynamic
+configuration of grafana, based on the topology changes in etcd. Some of these
+responsibilities would be:
+
+* monitoring-integration should store predefined json’s for dashboards, panels and alerts.
+* Keeping track of the topology changes in etcd to configure the grafana dashboards
+  to ensure that the correct nodes are part of the respective cluster dashboards.
+* Configuring grafana authentication and authorization (https://github.com/Tendrl/specifications/pull/205),
+  based on changes in the primary tendrl AA scheme.
+
+
+== Use Cases
+
+* Monitoring-integration should be able to configure grafana dashboard and panels
+  and alerts.
+* User should be able access grafana dashboard via tendrl user interface using
+  tendrl authentication.
+* Grafana alerting events should be notified via tendrl.
+
+== Proposed change
+
+=== Why monitoring-integration?
+
+Grafana is used for visualizing time series data in graphs. Based on the metrics
+from datasource it will create graphs and display. Grafana supports different kinds
+of http api's to configure its environment.
```

Comment:
@rishubhjain can you list down the API's that you have used to configure grafana?